File: pulse-go/options/logging.go
package options

type TimeFormat string

const (
	TimeFormatRFC3339     TimeFormat = "RFC3339"
	TimeFormatRFC3339Nano TimeFormat = "RFC3339Nano"
	TimeFormatKitchen     TimeFormat = "Kitchen"
	TimeFormatStamp       TimeFormat = "Stamp"
	TimeFormatCustom      TimeFormat = "Custom"
)

type LogOptions struct {
	Prefix          string     `json:"prefix"`           // Prefix string in log output
	ReportCaller    bool       `json:"report_caller"`    // Include caller info in logs
	ReportTimestamp bool       `json:"report_timestamp"` // Include timestamp in logs
	TimeFormatKey   TimeFormat `json:"time_format_key"`  // Named time format enum
	CustomFormat    string     `json:"custom_format"`    // Custom format if TimeFormatKey == TimeFormatCustom
	CallerOffset    int        `json:"caller_offset"`    // Adjust call depth for correct file/line display
}

type LoggingOptions struct {
	Log LogOptions `json:"log"` // Log options
}

File: pulse-rs/pulse/src/options/logging.rs
//! Logging configuration options.

use serde::{Deserialize, Serialize};

/// Time format options for log output.
#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub enum TimeFormat {
    #[default]
    RFC3339,
    RFC3339Nano,
    Kitchen,
    Stamp,
    Custom,
}

/// Log output configuration.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LogOptions {
    /// Prefix string in log output.
    #[serde(default)]
    pub prefix: String,
    /// Include caller info in logs.
    #[serde(default = "default_true")]
    pub report_caller: bool,
    /// Include timestamp in logs.
    #[serde(default = "default_true")]
    pub report_timestamp: bool,
    /// Named time format enum.
    #[serde(default)]
    pub time_format_key: TimeFormat,
    /// Custom format if time_format_key == Custom.
    #[serde(default)]
    pub custom_format: String,
    /// Adjust call depth for correct file/line display.
    #[serde(default = "default_caller_offset")]
    pub caller_offset: i32,
}

fn default_true() -> bool {
    true
}

fn default_caller_offset() -> i32 {
    3
}

impl Default for LogOptions {
    fn default() -> Self {
        Self {
            prefix: String::new(),
            report_caller: true,
            report_timestamp: true,
            time_format_key: TimeFormat::default(),
            custom_format: String::new(),
            caller_offset: 3,
        }
    }
}

/// Logging options container.
#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct LoggingOptions {
    /// Log options.
    #[serde(default)]
    pub log: LogOptions,
}
